neptune-core/src/p2p/protocol/messages.rs

```diff
@@ -0,0 +1,232 @@
+    /// Block proposal request
+    BlockProposalRequest(BlockProposalRequest),
+    /// Block proposal
+    BlockProposal(Box<Block>),
+    /// Transaction message
+    Transaction(Box<TransferTransaction>),
+    /// Transaction notification
+    TransactionNotification(TransactionNotification),
+    /// Transaction request
+    TransactionRequest(TransactionKernelId),
+    /// Peer list request
+    PeerListRequest,
+    /// Peer list response
+    PeerListResponse(Vec<(SocketAddr, u128)>),
+    /// Bye message
+    Bye,
+    /// Connection status
+    ConnectionStatus(TransferConnectionStatus),
+}
+
+impl PeerMessage {
+    /// Get the message type as a string
+    pub fn get_type(&self) -> &'static str {
+        match self {
+            PeerMessage::Handshake { .. } => "Handshake",
+            PeerMessage::Block(_) => "Block",
+            PeerMessage::BlockNotificationRequest => "BlockNotificationRequest",
+            PeerMessage::BlockNotification(_) => "BlockNotification",
+            PeerMessage::BlockRequestByHeight(_) => "BlockRequestByHeight",
+            PeerMessage::BlockRequestByHash(_) => "BlockRequestByHash",
+            PeerMessage::BlockRequestBatch(_) => "BlockRequestBatch",
+            PeerMessage::BlockResponseBatch(_) => "BlockResponseBatch",
+            PeerMessage::UnableToSatisfyBatchRequest => "UnableToSatisfyBatchRequest",
+            PeerMessage::SyncChallenge(_) => "SyncChallenge",
+            PeerMessage::SyncChallengeResponse(_) => "SyncChallengeResponse",
+            PeerMessage::BlockProposalNotification(_) => "BlockProposalNotification",
+            PeerMessage::BlockProposalRequest(_) => "BlockProposalRequest",
+            PeerMessage::BlockProposal(_) => "BlockProposal",
+            PeerMessage::Transaction(_) => "Transaction",
+            PeerMessage::TransactionNotification(_) => "TransactionNotification",
+            PeerMessage::TransactionRequest(_) => "TransactionRequest",
+            PeerMessage::PeerListRequest => "PeerListRequest",
+            PeerMessage::PeerListResponse(_) => "PeerListResponse",
+            PeerMessage::Bye => "Bye",
+            PeerMessage::ConnectionStatus(_) => "ConnectionStatus",
+        }
+    }
+
+    /// Check if message should be ignored during sync
+    pub fn ignore_during_sync(&self) -> bool {
+        match self {
+            PeerMessage::Block(_) => false,
+            PeerMessage::BlockNotification(_) => false,
+            PeerMessage::BlockResponseBatch(_) => false,
+            PeerMessage::SyncChallenge(_) => false,
+            PeerMessage::SyncChallengeResponse(_) => false,
+            _ => true, // Ignore other messages during sync
+        }
+    }
+
+    /// Check if message should be ignored when not syncing
+    pub fn ignore_when_not_sync(&self) -> bool {
+        match self {
+            PeerMessage::SyncChallenge(_) => true,
+            PeerMessage::SyncChallengeResponse(_) => true,
+            _ => false,
+        }
+    }
```

Comment:
Should sync challenges be ignored when you're not syncing? That doesn't sound right to me.